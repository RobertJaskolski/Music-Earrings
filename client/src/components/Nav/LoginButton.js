import React from "react";
import styled from "styled-components";
const Button = styled.button``;

function LoginButton() {
  return (
    <a href="Login">
      <Button>Connect with Spotify</Button>
    </a>
  );
}

export default LoginButton;
