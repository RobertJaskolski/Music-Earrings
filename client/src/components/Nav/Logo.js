import React from "react";
import logo from "../../images/logo.png";
import { Div } from "./style/style";

function Logo({ widthLogo, heightLogo }) {
  if (!widthLogo || !heightLogo) {
    return null;
  }

  return (
    <Div data-test='logoComponent'>
      <img
        data-test='logoIMG'
        src={logo}
        alt='Logo'
        width={widthLogo}
        height={heightLogo}
      />
    </Div>
  );
}

export default Logo;
