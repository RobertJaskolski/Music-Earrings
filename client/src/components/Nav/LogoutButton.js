import React from "react";
import styled from "styled-components";

const Button = styled.button`
  background-color: #1ed760;
  color: #fff;
  border: 1px solid #1db954;
  padding: 15px;
  border-radius: 5px;
  font-weight: bold;
  transition-duration: 0.5s;
  display: flex;
  align-items: center;
  &:hover {
    border-color: #1ed760;
    background-color: #1db954;
    cursor: pointer;
  }
`;
const Div = styled.div`
  display: flex;
  margin: 10px 10px 0px 0px;
  justify-content: flex-end;
  align-items: center;
`;

function LogoutButton() {
  return (
    <Div>
      <Button>Logout</Button>
    </Div>
  );
}

export default LogoutButton;
