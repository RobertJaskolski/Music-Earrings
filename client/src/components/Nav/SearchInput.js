import React from "react";
import TextField from "@material-ui/core/TextField";
import styled from "styled-components";
const Div = styled.div`
  display: flex;
  margin: 5px;
  justify-content: flex-start;
  align-items: center;
`;

function SearchInput() {
  return (
    <Div>
      <TextField label="Search artist, song or album..." fullWidth={true} />
    </Div>
  );
}

export default SearchInput;
