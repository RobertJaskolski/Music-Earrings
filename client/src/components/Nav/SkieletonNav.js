import React from "react";
import Skeleton from "@material-ui/lab/Skeleton";
import { Div } from "../../StyledComponents/GridElements";

export default function SkieletonNav() {
  return (
    <Div>
      <Skeleton variant='rect' width={80} height={50} />
    </Div>
  );
}
