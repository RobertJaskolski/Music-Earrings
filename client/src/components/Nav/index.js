export { default as SkieletonNav } from "./SkieletonNav";
export { default as LoginButton } from "./LoginButton";
export { default as Logo } from "./Logo";
export { default as LogoutButton } from "./LogoutButton";
export { default as SearchInput } from "./SearchInput";
