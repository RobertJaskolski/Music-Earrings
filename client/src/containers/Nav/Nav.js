import React from "react";
import { Grid } from "@material-ui/core";
import {
  LoginButton,
  Logo,
  SearchInput,
  LogoutButton,
  SkieletonNav,
} from "../../components/Nav";
import { connect } from "react-redux";
import { tokensActions } from "../../reducers/tokens";
import { authActions } from "../../reducers/auth";
import PropTypes from "prop-types";

const Nav = (props) => {
  const { logout, clear, loading, auth, userProfile } = props;
  const handleLogout = () => {
    logout();
    clear();
  };
  return (
    <Grid item>
      <nav>
        <Grid container spacing={1}>
          <Grid item sm={2}>
            <Logo widthLogo='75px' heightLogo='75px' />
          </Grid>
          <Grid item lg={7} md={6} sm={5}>
            <SearchInput />
          </Grid>
          <Grid item lg={3} md={4} sm={5}>
            {loading ? (
              <SkieletonNav />
            ) : auth ? (
              <LogoutButton
                logout={handleLogout}
                name={userProfile?.display_name || "Avatar"}
                imageURL={userProfile?.images[0].url || ""}
              />
            ) : (
              <LoginButton />
            )}
          </Grid>
        </Grid>
      </nav>
    </Grid>
  );
};
const mapDispatchToProps = (dispatch) => {
  return {
    logout: () => {
      dispatch(authActions.logout());
    },
    clear: () => {
      dispatch(tokensActions.clear());
    },
  };
};

Nav.propTypes = {
  logout: PropTypes.func,
  clear: PropTypes.func,
  auth: PropTypes.bool,
};

export default connect(null, mapDispatchToProps)(Nav);
