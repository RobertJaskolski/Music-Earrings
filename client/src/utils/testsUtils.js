import checkPropTypes from "check-prop-types";
export { default as testStore } from "../store";
export const findByDataTest = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );
  return propsErr;
};
