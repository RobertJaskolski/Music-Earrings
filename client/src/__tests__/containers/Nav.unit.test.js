import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Nav from '../../containers/Nav/Nav';
import { testStore } from '../../utils/testsUtils';

const store = (INIT_STATE = {}) => {
  const store = testStore(INIT_STATE);
  return store;
};

describe('Nav container', () => {
  test('should render component', () => {
    const wrapper = shallow(<Nav store={store()} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
