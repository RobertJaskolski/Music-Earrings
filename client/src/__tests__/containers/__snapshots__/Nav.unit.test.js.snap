// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav container should render component 1`] = `
<Nav
  MyAPIGetArtistAndTrack={[Function]}
  SpotifyGetArtistAndTrack={[Function]}
  clearTokens={[Function]}
  logout={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
